worlds/pokemon_emerald/__init__.py

```diff
@@ -328,16 +636,47 @@ def generate_basic(self) -> None:
                 "EVENT_VISITED_EVER_GRANDE_CITY"
             ])
 
-        self.free_fly_location_id = location_visited_event_to_id_map[fly_location_name]
+        self.free_fly_location_id = VISITED_EVENT_NAME_TO_ID[fly_location_name]
 
         free_fly_location_location = self.multiworld.get_location("FREE_FLY_LOCATION", self.player)
         free_fly_location_location.item = None
         free_fly_location_location.place_locked_item(self.create_event(fly_location_name))
 
+        # Set Marine Cave and Terra Cave entrances
+        terra_cave_location_name = self.random.choice([
+            "TERRA_CAVE_ROUTE_114_1",
+            "TERRA_CAVE_ROUTE_114_2",
+            "TERRA_CAVE_ROUTE_115_1",
+            "TERRA_CAVE_ROUTE_115_2",
+            "TERRA_CAVE_ROUTE_116_1",
+            "TERRA_CAVE_ROUTE_116_2",
+            "TERRA_CAVE_ROUTE_118_1",
+            "TERRA_CAVE_ROUTE_118_2"
+        ])
+
+        terra_cave_location_location = self.multiworld.get_location("TERRA_CAVE_LOCATION", self.player)
```

Comment:
Lol, to some degree I'm hoping the awkwardness actually calls attention to the distinction between the two meanings. I don't think there's really a clearer word to put in the location name. The variable only exists for these three lines to prevent extra calls to `get_location`. So I don't mind it. If it had any uses that weren't immediately next to its initialization, I would probably have changed it.